docs/TypeConverters/index.md

```diff
@@ -0,0 +1,148 @@
+# Type Converters
+
+Provides a unified way of converting types of values to other types, as well as for accessing values and supporting custom serialzation.
+
+## Using Type Converters
+
+Accessing a type converter retrieves the same instance always. 
+
+```csharp
+class SomeClass
+{
+	string m_ID;
+
+	int GetID()
+	{
+		auto converter = CF_TypeConverter.Get(string);
+
+		converter.SetString(m_ID);
+
+		return converter.GetInt();
+	}
+
+	void SetID(int id)
+	{
+		auto converter = CF_TypeConverter.Get(string);
+
+		converter.SetInt(id);
+
+		m_ID = converter.GetString();
+	}
+}
+```
+
+The variable name can be used to read and write the data. This is made with script reflection in mind. Just pass in the instance holding the variable and the variable name to the `Read` and `Write` methods. 
+
+```csharp
+class SomeClass
+{
+	string m_ID;
+
+	int GetID()
+	{
+		auto converter = CF_TypeConverter.Get(string);
```

Comment:
Would it make sense to offer a Get overload that accepts the string instance m_ID and just calls Get() with the typename of the instance? that way you could switch out m_ID from string to float or what ever and the rest of the code could still work

Reply:
Added `CF_TypeConverter::Get(instance, variableName)` in latest commit. But it is an O(n) operation - where `n` is the number of variables inside of `instance`